pkg/http/server.go

```diff
@@ -66,6 +65,44 @@ func (s *server) StartNonBlocking() {
 	}
 }
 
+func (s *server) useTracing(next fasthttp.RequestHandler) fasthttp.RequestHandler {
+	return diag.TracingHTTPMiddleware(s.tracingSpec, next)
+}
+
+func (s *server) useRouter() fasthttp.RequestHandler {
+	endpoints := s.api.APIEndpoints()
+	router := s.getRouter(endpoints)
+	return router.HandleRequest
+}
+
+func (s *server) useComponents(next fasthttp.RequestHandler) fasthttp.RequestHandler {
+	return s.pipeline.Apply(next)
+}
+
+func (s *server) useCors(next fasthttp.RequestHandler) fasthttp.RequestHandler {
+	origins := strings.Split(s.config.AllowedOrigins, ",")
+	corsHandler := s.getCorsHandler(origins)
+	return corsHandler.CorsMiddleware(next)
+}
+
+func (s *server) useProxy(next fasthttp.RequestHandler) fasthttp.RequestHandler {
+	return func(ctx *fasthttp.RequestCtx) {
+		var proto string
+		if ctx.IsTLS() {
+			proto = "https"
+		} else {
+			proto = "http"
+		}
+		ctx.Request.Header.Add("Forwarded",
+			fmt.Sprintf("by=%s;for=%s;host=%s;proto=%s",
+				ctx.LocalAddr(),
+				ctx.RemoteAddr(),
+				ctx.Host(),
+				proto))
```

Comment:
I think it should be better to add individual headers  `X-Forwarded-For`, `X-Forwarded-Host` and `X-Forwarded-Proto` so that upstream server don't have to parse `Forwarded` header and i think using individual headers is defacto-standard. Its just a suggestion and decision is upto you

Reply:
Yh I was a little torn which way to go - maybe we support both legacy and Forwarded (I was using NGINX as a reference: https://www.nginx.com/resources/wiki/start/topics/examples/forwarded/)?

Reply:
@yaron2 would you be happy supporting both `X-Forwarded-*` and `Forwarded` or do you have a preference?

Reply:
Supporting both seems right